src/main/java/ch/njol/skript/expressions/ExprAffectedEntities.java

```diff
@@ -58,20 +61,52 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 		return true;
 	}
 
-	@Nullable
 	@Override
-	protected LivingEntity[] get(Event e) {
-		if (e instanceof AreaEffectCloudApplyEvent)
-			return ((AreaEffectCloudApplyEvent) e).getAffectedEntities().toArray(new LivingEntity[0]);
+	protected LivingEntity @Nullable [] get(Event event) {
+		if (event instanceof AreaEffectCloudApplyEvent areaEvent)
+			return areaEvent.getAffectedEntities().toArray(new LivingEntity[0]);
 		return null;
 	}
 
-	@Nullable
 	@Override
-	public Iterator<? extends LivingEntity> iterator(Event e) {
-		if (e instanceof AreaEffectCloudApplyEvent)
-			return ((AreaEffectCloudApplyEvent) e).getAffectedEntities().iterator();
-		return super.iterator(e);
+	public @Nullable Iterator<? extends LivingEntity> iterator(Event event) {
+		if (event instanceof AreaEffectCloudApplyEvent areaEvent)
+			return areaEvent.getAffectedEntities().iterator();
+		return super.iterator(event);
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+        if (mode == ChangeMode.REMOVE_ALL)
+            return null;
+		return CollectionUtils.array(LivingEntity[].class);
```

Comment:
this should default to null and explicitly allow modes, rather than explicitly disallow, in the case a new change mode is added. A switch expression is good for this.